### PR TITLE
[FIX] pos_coupon: correct unit_price for rewardProduct

### PR DESCRIPTION
**Pull request by yosa-odoo**
Steps to reproduce:
- set a pricelist: discount, global, 10%
- set promotion program: free product
- In pos, select the product

Issue:
The product is deducted at its lst_price and not at the discounted price such as in `Sales`.

opw-2892748

### CODE REVIEW

**Review by agr-odoo — CHANGES_REQUESTED** [review]

File: addons/pos_coupon/static/src/js/coupon.js [hunk 1]
```diff
@@ -882,7 +882,7 @@ odoo.define('pos_coupon.pos', function (require) {
                     [
                         new Reward({
                             product: discountLineProduct,
-                            unit_price: -rewardProduct.lst_price,
+                            unit_price: -rewardProduct.get_price(this.pricelist),
```
[the remark this answers is not in the document]
Comment (yosa-odoo) [reply addons/pos_coupon/static/src/js/coupon.js]: I was not sure about setting the quantity, but after some more thought, I do agree. I just pushed the change and used the freeQuantity :)

**Review by agr-odoo — APPROVED** [review]

**Comment by yosa-odoo** [comment]
Hello @pimodoo,
Could you have a check at it or redirect me to someone else?
Cheers

**Comment by pimodoo** [comment]
@caburj does it looks good to you?

@yosa-odoo you should rouond what is given by get_price

**Review by caburj — APPROVED** [review]

File: addons/pos_coupon/static/src/js/coupon.js [hunk 2]
```diff
@@ -878,11 +878,13 @@ odoo.define('pos_coupon.pos', function (require) {
             } else {
                 const rewardProduct = this.pos.db.get_product_by_id(program.reward_product_id[0]);
                 const discountLineProduct = this.pos.db.get_product_by_id(program.discount_line_product_id[0]);
+                const price_unrounded = rewardProduct.get_price(this.pricelist, freeQuantity)
+                const price_rounded = round_decimals(price_unrounded, this.pos.currency.decimals || 2);
```
Comment (pimodoo) [inline addons/pos_coupon/static/src/js/coupon.js:882]: you can remove the "|| 2" and do it in one line.

**Comment by pimodoo** [comment]
@robodoo r+ rebase-ff

**Comment by yosa-odoo** [comment]
Thanks for the review!

**Comment by pimodoo** [comment]
@robodoo retry

**Pull request closed without merging**